packages/wallet/src/redux/protocols/indirect-funding/player-a/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,86 @@
+import * as scenarios from './scenarios';
+import { playerAReducer, initialize } from '../reducer';
+import { ProtocolStateWithSharedData } from '../../../../protocols';
+import { IndirectFundingState } from '../../state';
+import { SignedCommitment } from '../../../../../domain';
+import { getLastMessage } from '../../../../state';
+
+describe('happy-path scenario', () => {
+  const scenario = scenarios.happyPath;
+  describe('initializing', () => {
+    const { channelId, store, reply } = scenario.initialParams;
+    const initialState = initialize(channelId, store);
+
+    itTransitionsTo(initialState, 'AWaitForPreFundSetup1');
+    itSendsMessage(initialState, reply);
+  });
+
+  describe('when in WaitForPreFundL1', () => {
+    const { state, action } = scenario.waitForPreFundL1;
+    const updatedState = playerAReducer(state.state, state.store, action);
+
+    itTransitionsTo(updatedState, 'AWaitForDirectFunding');
+  });
+  describe('when in WaitForDirectFunding', () => {
+    const { state, action, reply } = scenario.waitForDirectFunding;
+    const updatedState = playerAReducer(state.state, state.store, action);
+
+    itTransitionsTo(updatedState, 'AWaitForLedgerUpdate1');
+    itSendsMessage(updatedState, reply);
+  });
+  describe('when in WaitForLedgerUpdate1', () => {
+    const { state, action, reply } = scenario.waitForLedgerUpdate1;
+    const updatedState = playerAReducer(state.state, state.store, action);
+
+    itTransitionsTo(updatedState, 'AWaitForPostFundSetup1');
+    itSendsMessage(updatedState, reply);
+  });
+  describe('when in WaitForPostFund0', () => {
+    const { state, action } = scenario.waitForPostFund1;
+    const updatedState = playerAReducer(state.state, state.store, action);
+
+    itTransitionsTo(updatedState, 'Success');
+  });
+});
+
+describe('ledger-funding-fails scenario', () => {
+  const scenario = scenarios.ledgerFundingFails;
+  describe('when in WaitForDirectFunding', () => {
+    const { state, action } = scenario.waitForDirectFunding;
+    const updatedState = playerAReducer(state.state, state.store, action);
+
+    itTransitionsTo(updatedState, 'Failure');
+  });
+});
+
+// -------
+// Helpers
+// -------
+type ReturnVal = ProtocolStateWithSharedData<IndirectFundingState>;
+
+function itTransitionsTo(state: ReturnVal, type: IndirectFundingState['type']) {
+  it(`transitions protocol state to ${type}`, () => {
+    expect(state.protocolState.type).toEqual(type);
+  });
+}
+
+function itSendsMessage(state: ReturnVal, message: SignedCommitment) {
```

Comment:
There's currently an `itSendsThisMessage` test helper, which could be used in place of this. I'm in favour of using a consistent test helper to expect side effects, rather than having each test suite define a bespoke helper.

The existing helper could be improved to meet the use case here (and its use updated in just a couple of places)